### ISSUE TITLE
Invalid JSON schema generated since a new version of typescript has been released

### ISSUE THREAD

**Issue by bertrandE** [YousefED/typescript-json-schema#420]
A new version of the typescript (4.3.2) NPM package has been released 20 hours ago (at the time of writing).
This project has a dependency on typescript (^4.2.3).

Since this new version of typescript, some  comments stopped working properly and generate invalid JSON schema (at least @minLength, @maxLength and @pattern).

To reproduce, take the following class for instance:

```
class MyClass {

    /**
     * @minItems 1
     * @maxItems 4
     */
    foo: Array<string>;

    /**
     * @pattern ^[0-9a-f]{6}$
     */
    bar: string;
}
```

The latest release generates the following invalid JSON schema:
```
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "properties": {
        "bar": {
            "pattern": [
                {
                    "kind": "text",
                    "text": "^[0-9a-f]{6}$"
                }
            ],
            "type": "string"
        },
        "foo": {
            "items": {
                "type": "string"
            },
            "maxItems": [
                {
                    "kind": "text",
                    "text": "4"
                }
            ],
            "minItems": [
                {
                    "kind": "text",
                    "text": "1"
                }
            ],
            "type": "array"
        }
    },
    "type": "object"
}
```
(note the **pattern**, **minItems** and **maxItems** properties which are arrays...)

Instead I was expecting
```
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "properties": {
        "bar": {
            "pattern": "^[0-9a-f]{6}$",
            "type": "string"
        },
        "foo": {
            "items": {
                "type": "string"
            },
            "maxItems": 4,
            "minItems": 1,
            "type": "array"
        }
    },
    "type": "object"
}
```
(which I generated by forcing typescript@4.2.4 instead of the latest release)

**Comment by domoritz** [comment]
I updated https://github.com/vega/ts-json-schema-generator already. The biggest change in TS was that comments are now parsed into a data structure instead of strings. 

Since ts doesn't follow semantic versioning, we should probably change the version match.

**Comment by domoritz** [comment]
Fixed in 0.50.1.
